app/src/main/java/com/voipgrid/vialer/CallActivity.java

```diff
@@ -202,6 +206,11 @@ protected void onResume() {
         }
 
         refreshAudioSourceButton();
+
+        if(!mNetworkUtil.isOnline()) {
+            mNetworkConnectivity.startNetworkAvailabilityActivity();
```

Comment:
You are on the correct lines here but when writing classes you should try and keep their responsibility to a single thing (it's called the Single Responsibility Principle if you want to read more). That class is a broadcast receiver that is responsible for listening when the network changes and then reacting, it shouldn't also be responsible for opening that activity from another place. 

I'd suggest creating a new static method on the NetworkAvailabilityActivity itself, then you could just call NetworkAvailabilityActivity.start(). You can also put the log line in there too.